goldendict.pro

```diff
@@ -642,6 +642,7 @@ TRANSLATIONS += locale/ru_RU.ts \
     locale/fi_FI.ts \
     locale/jb_JB.ts \
     locale/hi_IN.ts \
+    locale/ca_ES.ts \
```

Comment:
If the file is renamed to *ca_AD.ts*, the changes to *preferences.cc* woudn't be needed, I think.

Do Catalan dialects spoken in Andorra and Catalonia differ substantially?

Reply:
> If the file is renamed to _ca_AD.ts_, the changes to _preferences.cc_ woudn't be needed, I think.
> 
> Do Catalan dialects spoken in Andorra and Catalonia differ substantially?

About Andorra vs. Catalonia flag.

- Flags shouldn't be used to represent languages. It's a bad practice. More info [here](https://www.flagsarenotlanguages.com/blog/why-flags-do-not-represent-language/). Said that. It's common to find flags to represent languages in the wild (Goldendict in this case). It's a fact.
- Population from Andorra is about 80.000 inhabitants. Catalonia's population is about 8 millions. All Andorran population fits in "Camp Nou", FC Barcelona Stadium!
- Catalan is not a sovereign country, just like England, Scotland, or California. I'm not claiming such a thing. I'm only using flag from Catalonia to represent Catalan language. This flag is, by far, the expected one by Catalan and international speakers. Just google images "Catalan flag" or "Catalan language flag".
- Catalan language name takes its name from Catalonia territory.
- Some real examples in the wild using flag from Catalonia to represent Catalan language (**not** Andorra flag): [Booking.com](https://www.booking.com), [Ryanair](https://www.ryanair.com), [LanguageTool](https://languagetool.org/ca), [Learn Catalan from Spanish Dulingo course](https://es.duolingo.com/course/ca/es/Aprender-catal%C3%A1n), [phpBB language modules](https://www.phpbb.com/languages/). 


So, If flags are used by Goldendict to represent languages and UI translations, then I request to use flag from Catalonia to represent Catalan language for dictionaries and UI translations.

Reply:
Right, the issue of using flags to represent languages has been raised before: #1330. Convenient to some, offending to others, no easy way to please everyone.

Your arguments in favor of the Catalan flag are convincing. Recently I have made some tentative country/language code changes locally to support different Belarusian language variants. I'll try to improve your C++ changes and hopefully make them more general.

Reply:
How about renaming the file to *ct_ES.ts*, renaming *catalonia.png* to *ct.png*, replacing "catalonia" with "ct" in *language.cc* and in *flags.qrc*, reverting the changes in *preferences.cc*?

Catalonia's ISO 3166 code is **ES-CT**. According to https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2, the **CT** country code is deleted and free for reassignment (formerly "Canton and Enderbury Islands", a territory I haven't heard of before :)).

Reply:
Well, using ca_CT.ts (ca language code followed by CT fake-country code) is good enough for me. I didn't want to use a country code for Catalonia, but if you feel better with this approach, I have not any problem ;)

Reply:
Right, my mistake, I meant ca_CT.ts. We already have a fake Lojban country code JB, so I prefer this approach to the custom replacement in *preferences.cc*.

Reply:
You forgot to rename the icon in *flags.qrc*.

Reply:
Thanks!

I tried to squash commits, without luck. I'm not a programmer guy. Please, can you point me how this can be done? Thanks

Reply:
https://garrytrinder.github.io/2020/03/squashing-commits-using-interactive-rebase

Basically, `git rebase -i @~5`, where 5 is the number of commits to squash. Then replace all `pick` except for the first one with `squash`. All 5 commit messages will be combined in one text and an editor will open letting you edit the new single commit message. Once done, verify nothing has changed, e.g. `git diff origin/master` should output nothing (assuming your remote is called `origin` locally). Then `git push -f origin master:master` (again assuming your remote is called `origin` locally).

Reply:
Thanks! I hope is done now :)

Reply:
Looks reasonable. I'll recheck and maybe approve tomorrow.